pkg/client/interface.go

```diff
@@ -18,8 +31,21 @@ import (
 	"pocket/pkg/observable"
 )
 
+// TxClient provides an interface for initiating transactions on a blockchain.
+// It offers a singular, streamlined method to sign and broadcast multiple messages
+// as part of a transaction. This interface abstracts the actual sending of a
+// transaction, while more intricate functionalities like building, encoding,
+// or querying transactions are likely to be handled by other components,
+// such as the TxContext.
+type TxClient interface {
+	SignAndBroadcast(
```

Comment:
Unclear if an error is returned if one of or all of msgs error

Reply:
I think the answer to this is comes from cosmos-sdk specific behavior. I've rephrased the comments to better reflect where things are cosmos-sdk specific and where things are intended to be used for Pocket protocol purposes (e.g. implementing an off-chain actor).